infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/GitUserProfileProvisioner.java

```diff
@@ -58,61 +62,87 @@
   private static final String CONFIG_MAP_VOLUME_NAME = "gitconfigvolume";
 
   private PreferenceManager preferenceManager;
+  private UserManager userManager;
 
   @Inject
-  public GitUserProfileProvisioner(PreferenceManager preferenceManager) {
+  public GitUserProfileProvisioner(PreferenceManager preferenceManager, UserManager userManager) {
     this.preferenceManager = preferenceManager;
+    this.userManager = userManager;
   }
 
   @Override
   public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       throws InfrastructureException {
+
     try {
-      doInternalProvision(k8sEnv, identity);
+      getUserFromPreferences(
+          (name, email) -> {
+            if (isNullOrEmpty(name) && isNullOrEmpty(email)) {
```

Comment:
It might be clearer to invert this logic:
```java
if (!isNullOrEmpty(name) || !isNullOrEmpty(email)) {
  prepareAndProvisionGitConfiguration(name, email, k8sEnv, identity);
} else {
  // provision from user manager
}
```

Reply:
I think, approach, provided by @metlos will be clearer. I'll implement it.